src/impurity_init.ipp

```diff
@@ -417,7 +417,7 @@ void HybridizationSimulation<IMP_MODEL>::read_two_time_correlation_functions() {
     return;
   }
 
-  std::ifstream infile_f(boost::lexical_cast<std::string>(par[fname_key]).c_str());
+  std::ifstream infile_f(boost::lexical_cast<std::string>(par[fname_key].template as<std::string>()));
```

Comment:
The same comment as above: the lexical cast was erroneous here; it is static cast that is needed:
```c++
std::ifstream infile_f(par[fname_key].as<std::string>());
```
or
```c++
std::ifstream infile_f(std::string(par[fname_key]));
```